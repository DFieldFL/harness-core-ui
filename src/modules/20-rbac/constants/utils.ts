export enum RbacResourceGroupTypes {
  DYNAMIC_RESOURCE_SELECTOR = 'DynamicResourceSelector'
}
