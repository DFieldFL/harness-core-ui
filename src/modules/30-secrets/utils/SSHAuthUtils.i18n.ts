export default {
  typeText: 'Text',
  typeFile: 'File',
  typeSSH: 'Execution Credential',

  optionKey: 'Username/SSH Key',
  optionKeypath: 'Username/SSH Key File Path',
  optionPassword: 'Username/Password'
}
