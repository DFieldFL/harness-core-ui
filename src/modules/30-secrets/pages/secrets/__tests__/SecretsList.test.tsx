import React from 'react'
import { act, fireEvent, getByText, queryByText, render, RenderResult, waitFor } from '@testing-library/react'
import { noop } from 'lodash-es'

import { findDialogContainer, findPopoverContainer, TestWrapper } from '@common/utils/testUtils'
import type { PageSecretResponseWrapper } from 'services/cd-ng'
import SecretsList from '../views/SecretsListView/SecretsList'

import mockData from './secretsListMock.json'

jest.mock('react-timeago', () => () => 'dummy date')

describe('Secrets List', () => {
  let container: HTMLElement
  let getAllByText: RenderResult['getAllByText']
  beforeEach(async () => {
    const renderObj = render(
      <TestWrapper path="/account/:accountId/resources/secrets" pathParams={{ accountId: 'dummy' }}>
        <SecretsList secrets={mockData.data as PageSecretResponseWrapper} gotoPage={noop} />
      </TestWrapper>
    )
    container = renderObj.container
    getAllByText = renderObj.getAllByText
  }),
    test('render', async () => {
      expect(container).toMatchSnapshot()
      expect(container.querySelectorAll('div.row').length).toBe(4)
    }),
    test('Edit', async () => {
      const menu = container?.querySelectorAll("[id='Options_svg__a']")[0]
      fireEvent.click(menu!)
      const popover = findPopoverContainer()
      const edit = getByText(popover as HTMLElement, 'Edit')
      await act(async () => {
        fireEvent.click(edit)
      })
      expect(container).toMatchSnapshot()
    }),
    test('Delete', async () => {
      const menu = container?.querySelectorAll("[id='Options_svg__a']")[0]
      fireEvent.click(menu!)
      const popover = findPopoverContainer()
      const deleteButton = getByText(popover as HTMLElement, 'Delete')
      await act(async () => {
        fireEvent.click(deleteButton)
        await waitFor(() => getByText(document.body, 'Delete Secret'))
        const form = findDialogContainer()
        expect(form).toBeTruthy()
        const deleteBtn = queryByText(form as HTMLElement, 'Delete')
        fireEvent.click(deleteBtn!)
      })
      expect(container).toMatchSnapshot()
    })
  test('Verify Connection', async () => {
    const testConnection = getAllByText('TEST CONNECTION')[0]
    let form = findDialogContainer()
    await act(async () => {
      fireEvent.click(testConnection)
      await waitFor(() => getByText(document.body, 'Test Connection'))
    })
    form = findDialogContainer()
    expect(form).toBeTruthy()
    await act(async () => {
      fireEvent.click(form?.querySelector('[icon="cross"]')!)
    })
    form = findDialogContainer()
    expect(form).not.toBeTruthy()
  })
})
