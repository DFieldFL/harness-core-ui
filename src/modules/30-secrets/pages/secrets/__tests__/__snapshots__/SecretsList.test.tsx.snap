// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secrets List render 1`] = `
<div>
  <div
    class="table table"
  >
    <div
      class="header"
      role="row"
    >
      <div
        class="cell"
        colspan="1"
        role="columnheader"
        style="width: 30%;"
        title="Toggle SortBy"
      >
        SECRET
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="cell"
        colspan="1"
        role="columnheader"
        style="width: 25%;"
        title="Toggle SortBy"
      >
        DETAILS
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="cell"
        colspan="1"
        role="columnheader"
        style="width: 20%;"
        title="Toggle SortBy"
      >
        LAST ACTIVITY
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="cell"
        colspan="1"
        role="columnheader"
        style="width: 20%;"
      >
        
      </div>
      <div
        class="cell"
        colspan="1"
        role="columnheader"
        style="width: 5%;"
      >
        
      </div>
    </div>
    <div
      class="row card clickable"
      role="row"
    >
      <div
        class="cell"
        role="cell"
        style="width: 30%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws StyledProps---margin-top-xsmall--33hS StyledProps---margin-right-small--Uojv"
          >
            <svg
              height="28"
              viewBox="0 0 89 73"
              width="28"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                transform="translate(-1 -1)"
              >
                <path
                  d="M86 0H5a5 5 0 00-5 5v65a5 5 0 005 5h81a5 5 0 005-5V5a5 5 0 00-5-5z"
                  fill="#1C1C28"
                  fill-rule="nonzero"
                />
                <path
                  d="M5 2h81a3 3 0 013 3v65a3 3 0 01-3 3H5a3 3 0 01-3-3V5a3 3 0 013-3z"
                  fill="#FFE981"
                  fill-rule="nonzero"
                />
                <path
                  d="M69.5 16.5v2H.5v-2zm16.5 0v2H74v-2zM7 6a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1zm6 0a1 1 0 011 1v5a1 1 0 01-2 0V7a1 1 0 011-1z"
                  fill="#1C1C28"
                  fill-rule="nonzero"
                />
                <path
                  d="M16.2 52.8c.633 0 1.167-.217 1.6-.65l8.4-8.4c.067-.067.1-.117.1-.15.167-.167.3-.383.4-.65.1-.267.15-.517.15-.75v-.05c0-.333-.067-.633-.2-.9-.133-.3-.283-.517-.45-.65l-8.4-8.4c-.2-.2-.442-.35-.725-.45a3.922 3.922 0 00-.875-.2c-.567-.033-1.083.183-1.55.65-.233.2-.408.45-.525.75a2.34 2.34 0 00-.175.85c0 .6.233 1.133.7 1.6l4.75 4.75c.567.567.85 1.242.85 2.025 0 .783-.283 1.458-.85 2.025l-.975.975c-.317.317-.642.625-.975.925-.467.467-.933.942-1.4 1.425-.467.483-.933.958-1.4 1.425-.4.4-.633.933-.7 1.6 0 .3.067.583.2.85s.3.517.5.75c.167.2.4.358.7.475.3.117.583.175.85.175z"
                  fill="#22272D"
                  fill-rule="nonzero"
                />
                <rect
                  fill="#1C1C28"
                  height="3"
                  rx="1.5"
                  width="40"
                  x="28"
                  y="50"
                />
              </g>
            </svg>
          </span>
          <div
            class="Layout---vertical--3Szv StyledProps---main--1Wws"
          >
            <div
              class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
              style="width: 230px;"
            >
              <p
                class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
                style="--text-line-clamp: 1;"
              >
                kerb5
              </p>
            </div>
            <p
              class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
              style="--text-line-clamp: 1; width: 230px;"
            >
              kerb5
            </p>
          </div>
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 25%;"
      >
        <p
          class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
        >
          Execution Credential
        </p>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <div
          class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws"
          >
            <svg
              height="16"
              width="16"
            >
              <path
                d="M8 0C3.538.036-.035 3.644 0 8.053.036 12.427 3.61 16 8 16h.054c2.133-.018 4.142-.853 5.635-2.382 1.493-1.53 2.329-3.538 2.311-5.671C15.965 3.573 12.391 0 8 0zm5.262 8.836h-1.635a.52.52 0 01-.48-.338l-.836-2.276-1.653 5.245a.49.49 0 01-.462.355H8.16a.51.51 0 01-.462-.302L5.796 7.253l-.907 1.351a.5.5 0 01-.427.232H2.667a.516.516 0 010-1.032h1.511L5.44 5.902a.513.513 0 01.48-.23.488.488 0 01.427.301L8.07 9.867l1.725-5.423a.512.512 0 01.48-.355c.23 0 .426.124.498.338l1.208 3.36h1.28c.285 0 .516.23.516.515a.531.531 0 01-.516.534z"
                fill="#1D76FF"
              />
            </svg>
          </span>
          dummy date
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <button
          class="bp3-button StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws StyledProps---font-small--1qQz"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            TEST CONNECTION
          </span>
        </button>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 5%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
          style="justify-content: flex-end;"
        >
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps---main--1Wws"
                >
                  <svg
                    height="24"
                    width="24"
                  >
                    <defs>
                      <path
                        d="M0 0h24v24H0z"
                        id="Options_svg__a"
                      />
                    </defs>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <mask
                        fill="#fff"
                        id="Options_svg__b"
                      >
                        <use
                          xlink:href="#Options_svg__a"
                        />
                      </mask>
                      <path
                        d="M17.5 13.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm-11 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5.5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
                        mask="url(#Options_svg__b)"
                        stroke="#25A6F7"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        transform="rotate(90 12 12)"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="row card clickable"
      role="row"
    >
      <div
        class="cell"
        role="cell"
        style="width: 30%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws StyledProps---margin-top-xsmall--33hS StyledProps---margin-right-small--Uojv"
          >
            <svg
              height="28"
              width="28"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                opacity="0.6"
              >
                <circle
                  cx="14"
                  cy="14"
                  fill="#D0E2FF"
                  r="14"
                />
                <path
                  d="M16.488 6.533a4.977 4.977 0 00-4.977 4.978c0 .445.064.874.174 1.285l-5.152 5.151v3.519h3.52l5.151-5.151c.41.11.84.174 1.285.174a4.978 4.978 0 000-9.956zm-6.95 13.69l-1.76-.002v-1.759l4.426-4.426a4.997 4.997 0 001.76 1.76l-4.426 4.426zm6.95-4.979a3.733 3.733 0 110-7.466 3.732 3.732 0 013.734 3.733 3.732 3.732 0 01-3.734 3.733zm2.125-4.613a1.243 1.243 0 11-2.489 0 1.245 1.245 0 112.489 0z"
                  fill="#0042A9"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </span>
          <div
            class="Layout---vertical--3Szv StyledProps---main--1Wws"
          >
            <div
              class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
              style="width: 230px;"
            >
              <p
                class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
                style="--text-line-clamp: 1;"
              >
                yaml2
              </p>
            </div>
            <p
              class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
              style="--text-line-clamp: 1; width: 230px;"
            >
              yaml2
            </p>
          </div>
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 25%;"
      >
        <p
          class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
          style="--text-line-clamp: 1; width: 230px;"
        >
          harnessSecretManager
        </p>
        <p
          class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
        >
          Text
        </p>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <div
          class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws"
          >
            <svg
              height="16"
              width="16"
            >
              <path
                d="M8 0C3.538.036-.035 3.644 0 8.053.036 12.427 3.61 16 8 16h.054c2.133-.018 4.142-.853 5.635-2.382 1.493-1.53 2.329-3.538 2.311-5.671C15.965 3.573 12.391 0 8 0zm5.262 8.836h-1.635a.52.52 0 01-.48-.338l-.836-2.276-1.653 5.245a.49.49 0 01-.462.355H8.16a.51.51 0 01-.462-.302L5.796 7.253l-.907 1.351a.5.5 0 01-.427.232H2.667a.516.516 0 010-1.032h1.511L5.44 5.902a.513.513 0 01.48-.23.488.488 0 01.427.301L8.07 9.867l1.725-5.423a.512.512 0 01.48-.355c.23 0 .426.124.498.338l1.208 3.36h1.28c.285 0 .516.23.516.515a.531.531 0 01-.516.534z"
                fill="#1D76FF"
              />
            </svg>
          </span>
          dummy date
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <p
          class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---intent-warning--2N1t"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon bp3-icon-warning-sign StyledProps---main--1Wws StyledProps---padding-right-xsmall--3D04"
            icon="warning-sign"
          >
            <svg
              data-icon="warning-sign"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                warning-sign
              </desc>
              <path
                d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          Incomplete Secret
        </p>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 5%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
          style="justify-content: flex-end;"
        >
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps---main--1Wws"
                >
                  <svg
                    height="24"
                    width="24"
                  >
                    <defs>
                      <path
                        d="M0 0h24v24H0z"
                        id="Options_svg__a"
                      />
                    </defs>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <mask
                        fill="#fff"
                        id="Options_svg__b"
                      >
                        <use
                          xlink:href="#Options_svg__a"
                        />
                      </mask>
                      <path
                        d="M17.5 13.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm-11 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5.5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
                        mask="url(#Options_svg__b)"
                        stroke="#25A6F7"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        transform="rotate(90 12 12)"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="row card clickable"
      role="row"
    >
      <div
        class="cell"
        role="cell"
        style="width: 30%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws StyledProps---margin-top-xsmall--33hS StyledProps---margin-right-small--Uojv"
          >
            <svg
              height="28"
              width="28"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                opacity="0.6"
              >
                <circle
                  cx="14"
                  cy="14"
                  fill="#D0E2FF"
                  r="14"
                />
                <path
                  d="M16.488 6.533a4.977 4.977 0 00-4.977 4.978c0 .445.064.874.174 1.285l-5.152 5.151v3.519h3.52l5.151-5.151c.41.11.84.174 1.285.174a4.978 4.978 0 000-9.956zm-6.95 13.69l-1.76-.002v-1.759l4.426-4.426a4.997 4.997 0 001.76 1.76l-4.426 4.426zm6.95-4.979a3.733 3.733 0 110-7.466 3.732 3.732 0 013.734 3.733 3.732 3.732 0 01-3.734 3.733zm2.125-4.613a1.243 1.243 0 11-2.489 0 1.245 1.245 0 112.489 0z"
                  fill="#0042A9"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </span>
          <div
            class="Layout---vertical--3Szv StyledProps---main--1Wws"
          >
            <div
              class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
              style="width: 230px;"
            >
              <p
                class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
                style="--text-line-clamp: 1;"
              >
                filen1
              </p>
            </div>
            <p
              class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
              style="--text-line-clamp: 1; width: 230px;"
            >
              filen1
            </p>
          </div>
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 25%;"
      >
        <p
          class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
          style="--text-line-clamp: 1; width: 230px;"
        >
          vault1
        </p>
        <p
          class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
        >
          File
        </p>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <div
          class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws"
          >
            <svg
              height="16"
              width="16"
            >
              <path
                d="M8 0C3.538.036-.035 3.644 0 8.053.036 12.427 3.61 16 8 16h.054c2.133-.018 4.142-.853 5.635-2.382 1.493-1.53 2.329-3.538 2.311-5.671C15.965 3.573 12.391 0 8 0zm5.262 8.836h-1.635a.52.52 0 01-.48-.338l-.836-2.276-1.653 5.245a.49.49 0 01-.462.355H8.16a.51.51 0 01-.462-.302L5.796 7.253l-.907 1.351a.5.5 0 01-.427.232H2.667a.516.516 0 010-1.032h1.511L5.44 5.902a.513.513 0 01.48-.23.488.488 0 01.427.301L8.07 9.867l1.725-5.423a.512.512 0 01.48-.355c.23 0 .426.124.498.338l1.208 3.36h1.28c.285 0 .516.23.516.515a.531.531 0 01-.516.534z"
                fill="#1D76FF"
              />
            </svg>
          </span>
          dummy date
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      />
      <div
        class="cell"
        role="cell"
        style="width: 5%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
          style="justify-content: flex-end;"
        >
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps---main--1Wws"
                >
                  <svg
                    height="24"
                    width="24"
                  >
                    <defs>
                      <path
                        d="M0 0h24v24H0z"
                        id="Options_svg__a"
                      />
                    </defs>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <mask
                        fill="#fff"
                        id="Options_svg__b"
                      >
                        <use
                          xlink:href="#Options_svg__a"
                        />
                      </mask>
                      <path
                        d="M17.5 13.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm-11 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5.5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
                        mask="url(#Options_svg__b)"
                        stroke="#25A6F7"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        transform="rotate(90 12 12)"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="row card clickable"
      role="row"
    >
      <div
        class="cell"
        role="cell"
        style="width: 30%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws StyledProps---margin-top-xsmall--33hS StyledProps---margin-right-small--Uojv"
          >
            <svg
              height="28"
              width="28"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                opacity="0.6"
              >
                <circle
                  cx="14"
                  cy="14"
                  fill="#D0E2FF"
                  r="14"
                />
                <path
                  d="M16.488 6.533a4.977 4.977 0 00-4.977 4.978c0 .445.064.874.174 1.285l-5.152 5.151v3.519h3.52l5.151-5.151c.41.11.84.174 1.285.174a4.978 4.978 0 000-9.956zm-6.95 13.69l-1.76-.002v-1.759l4.426-4.426a4.997 4.997 0 001.76 1.76l-4.426 4.426zm6.95-4.979a3.733 3.733 0 110-7.466 3.732 3.732 0 013.734 3.733 3.732 3.732 0 01-3.734 3.733zm2.125-4.613a1.243 1.243 0 11-2.489 0 1.245 1.245 0 112.489 0z"
                  fill="#0042A9"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </span>
          <div
            class="Layout---vertical--3Szv StyledProps---main--1Wws"
          >
            <div
              class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
              style="width: 230px;"
            >
              <p
                class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
                style="--text-line-clamp: 1;"
              >
                yaml1
              </p>
            </div>
            <p
              class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
              style="--text-line-clamp: 1; width: 230px;"
            >
              yaml1
            </p>
          </div>
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 25%;"
      >
        <p
          class="StyledProps---font--2ipy Text---lineclamp--qeDh StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
          style="--text-line-clamp: 1; width: 230px;"
        >
          harnessSecretManager
        </p>
        <p
          class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
        >
          Text
        </p>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      >
        <div
          class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws"
          >
            <svg
              height="16"
              width="16"
            >
              <path
                d="M8 0C3.538.036-.035 3.644 0 8.053.036 12.427 3.61 16 8 16h.054c2.133-.018 4.142-.853 5.635-2.382 1.493-1.53 2.329-3.538 2.311-5.671C15.965 3.573 12.391 0 8 0zm5.262 8.836h-1.635a.52.52 0 01-.48-.338l-.836-2.276-1.653 5.245a.49.49 0 01-.462.355H8.16a.51.51 0 01-.462-.302L5.796 7.253l-.907 1.351a.5.5 0 01-.427.232H2.667a.516.516 0 010-1.032h1.511L5.44 5.902a.513.513 0 01.48-.23.488.488 0 01.427.301L8.07 9.867l1.725-5.423a.512.512 0 01.48-.355c.23 0 .426.124.498.338l1.208 3.36h1.28c.285 0 .516.23.516.515a.531.531 0 01-.516.534z"
                fill="#1D76FF"
              />
            </svg>
          </span>
          dummy date
        </div>
      </div>
      <div
        class="cell"
        role="cell"
        style="width: 20%;"
      />
      <div
        class="cell"
        role="cell"
        style="width: 5%;"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws"
          style="justify-content: flex-end;"
        >
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps---main--1Wws"
                >
                  <svg
                    height="24"
                    width="24"
                  >
                    <defs>
                      <path
                        d="M0 0h24v24H0z"
                        id="Options_svg__a"
                      />
                    </defs>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <mask
                        fill="#fff"
                        id="Options_svg__b"
                      >
                        <use
                          xlink:href="#Options_svg__a"
                        />
                      </mask>
                      <path
                        d="M17.5 13.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm-11 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5.5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
                        mask="url(#Options_svg__b)"
                        stroke="#25A6F7"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        transform="rotate(90 12 12)"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="Layout---horizontal--3rmQ StyledProps---main--1Wws Pagination---container--3ype StyledProps---padding-top-medium--29t2 StyledProps---padding-bottom-medium--3R7v StyledProps---flex--2Jz8 StyledProps---flex-align-center-center--2AmX StyledProps---flex-distribution-space-between--1TyN"
    >
      <span
        class="StyledProps---font--2ipy StyledProps---main--1Wws Pagination---text--30fh StyledProps---inline--1f52"
      >
        Showing 
        <strong>
          1
        </strong>
         - 
        <strong>
          4
        </strong>
         of
         
        <strong>
          30
        </strong>
      </span>
      <div
        class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
      >
        <button
          class="bp3-button bp3-disabled StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws StyledProps---intent-primary--2Xgi"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            1
          </span>
        </button>
        <button
          class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            2
          </span>
        </button>
        <button
          class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            3
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
